### PR TITLE
Add support for blob batch

### PR DESCRIPTION
**Pull request by EmmaZhu**
Thanks for contribution! Please go through following checklist before sending PR.

### PR Branch Destination

- For Azurite V3, please send PR to `main` branch.
- For legacy Azurite V2, please send PR to `legacy-dev` branch.

### Always Add Test Cases

Make sure test cases are added to cover the code change.

### Add Change Log

Add change log for the code change in `Upcoming Release` section in `ChangeLog.md`.

### Development Guideline

Please go to CONTRIBUTION.md for steps about setting up development environment and recommended Visual Studio Code extensions.

Manually tested:
1. subrequest count exceed the limition (> 256)
2. Different operation method in one batch request
3. No id in one of subrequest

Batch request support all of share key/SAS token/AAD token credentials in both of service and container scope batch request.

### CODE REVIEW

**Comment by EmmaZhu** [comment]
Added test case for different SAS token resource scope for container endpoint/service endpoint.

**Comment by EmmaZhu** [comment]
Added test cases for container endpoint where subrequest sent to blobs under a different container.

**Comment by EmmaZhu** [comment]
Confirm about the REST API credential description about container endpoint and service endpoint

File: src/blob/handlers/BlobBatchHandler.ts [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+
+    // "DELETE /container166063791875402779/blob0 HTTP/1.1"
+    ++lineIndex;
+    const operationInfos = requestLines[lineIndex].split(" ");
+    if (operationInfos.length < 3) throw new Error("Bad request");
+    const url = `${request.getEndpoint()}${operationInfos[1]}`;
+    const method = operationInfos[0] as HttpMethod;
+    if (blobBatchSubRequests.length !== 0)
+    {
+      if (method !== blobBatchSubRequests[0].getMethod()) throw new Error("Not support different method");
```
Comment (EmmaZhu) [inline src/blob/handlers/BlobBatchHandler.ts:338]: Should also compare endpoint pattern and comtype.

Comment (EmmaZhu) [reply src/blob/handlers/BlobBatchHandler.ts:338]: Add comments to pay attention on copy request (if need to support batch copy in the future)

**Comment by EmmaZhu** [comment]
Add comments in blob.md about the change of adding container_batch operation.

**Comment by EmmaZhu** [comment]
Limitation of subrequest count

**Comment by EmmaZhu** [comment]
Log subrequest ids in the batch method.

**Comment by EmmaZhu** [comment]
Reference to test cases for table batch

File: src/blob/handlers/BlobBatchHandler.ts (hunk 1 above)
Comment (EmmaZhu) [reply src/blob/handlers/BlobBatchHandler.ts]: Updated

**Review by blueww — APPROVED** [review]

File: tests/blob/apis/blobbatch.test.ts [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+  AnonymousCredential,
+  ContainerSASPermissions
+} from "@azure/storage-blob";
+import assert from "assert";
+import { configLogger } from "../../../src/common/Logger";
+import BlobTestServerFactory from "../../BlobTestServerFactory";
+import { EMULATOR_ACCOUNT_KEY, EMULATOR_ACCOUNT_NAME, getUniqueName } from "../../testutils";
+
+// Set true to enable debug log
+configLogger(true);
```
Comment (XiaoningLiu) [inline tests/blob/apis/blobbatch.test.ts:18]: Set it to "false" before submitting PR.

**Review by XiaoningLiu — APPROVED** [review]

**Pull request merged**